### PR TITLE
Add regression test for #373

### PR DESCRIPTION
**Pull request by Synthetica9**
https://github.com/haskell-nix/hnix/pull/373

cc @domenkozar

### CODE REVIEW

File: tests/THTests.hs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import           Test.Tasty
+import           Test.Tasty.HUnit
+import           Test.Tasty.TH
+
+-- Regression test for #373
+case_regression_373 :: Assertion
+case_regression_373 = do
+  freeVarsEqualTest "{ inherit a; }" ["a"]
+  freeVarsEqualTest "rec {inherit a; }" ["a"]
+  freeVarsEqualTest "let inherit a; in { }" ["a"]
```
Comment (infinisil) [inline tests/THTests.hs]: There is some fancy TemplateHaskell thing to have Nix expressions directly without going through strings, but I don't know how it works

**Comment by Synthetica9** [comment]
And neither do I, I mostly looked what the other test files did

EDIT: unless you're talking about QuasiQuotes (which the TH file is mostly about), which could be used, yes.

**Comment by domenkozar** [comment]
the test can just go to tests/EvalTests.hs

**Comment by Synthetica9** [comment]
> the test can just go to tests/EvalTests.hs

See 944ceda

**Comment by infinisil** [comment]
Ohhh, this is a PR against my fork! Realized only just now

**Comment by Synthetica9** [comment]
Yeah, Github should honestly have a better system to propose commits to
PR's for non-maintainers (or am I missing something?)

**Comment by infinisil** [comment]
@Synthetica9 There is! https://help.github.com/articles/incorporating-feedback-in-your-pull-request/

**Comment by Synthetica9** [comment]
Yeah, they should probably flesh that out a bit more, this commit for instance couldn't have been done that way, because I was editing files that weren't in the previous PR. It's a step in the right direction, but it needs a bit more fleshing-out

**Comment by infinisil** [comment]
Oh right

**Pull request merged**